### PR TITLE
LocalScanner: Fix-up the creation of missing file archives

### PR DESCRIPTION
**Pull request by fviernau**
Previously, mapNotNullTo() added the mapped elements to the wrong set,
and thus `missingArchives` was guaranteed to always remain empty.

Trivially fix this by assigning the variable properly. This is a fix-up
of 90de765.

### CODE REVIEW

File: scanner/src/main/kotlin/LocalScanner.kt [hunk 1]
```diff
@@ -271,14 +271,14 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
...
-                    val missingArchives = HashSet<Provenance>()
 
-                    storedResults.mapNotNullTo(mutableSetOf()) { result ->
+                    val missingArchives = storedResults.mapNotNullTo(mutableSetOf()) { result ->
                         result.provenance.takeUnless { archiver.hasArchive("$packagePath/${it.hash()}") }
                     }
 
                     if (missingArchives.isNotEmpty()) {
                         val downloadResult = Downloader.download(pkg, downloadDirectory.resolve(packagePath))
+
```
Comment (sschuberth) [inline scanner/src/main/kotlin/LocalScanner.kt:281]: Nit: Unintended change?

**Comment by sschuberth** [comment]
> Trivially fix this by assigning the variable properly.

Thanks for the fix, BTW!

File: scanner/src/main/kotlin/LocalScanner.kt (hunk 1 above)
Comment (fviernau) [reply scanner/src/main/kotlin/LocalScanner.kt:281]: nope, it was intended. See the commit message ;-)

Comment (sschuberth) [reply scanner/src/main/kotlin/LocalScanner.kt:281]: Oops, sorry, as the first discussion post looked like a commit message, I thought there's only one commit in the PR!

**Review by sschuberth — APPROVED** [review]

**Pull request merged**